src/scenes/Activities/scenes/EventHistory/index.js

```diff
@@ -117,90 +127,110 @@ class EventHistory extends React.Component {
   }
 
   getTableHeader() {
-    const { order, orderBy } = this.state;
-
     const headerCols = [
       {
-        id: 'time',
+        id: 'createdTime',
         name: 'str.time',
         nameDefault: 'Time',
         numeric: false,
+        sortable: true,
       },
       {
         id: 'type',
         name: 'str.type',
         nameDefault: 'Type',
         numeric: false,
+        sortable: true,
       },
       {
         id: 'name',
         name: 'str.name',
         nameDefault: 'Name',
         numeric: false,
+        sortable: false,
       },
       {
         id: 'token',
         name: 'str.token',
         nameDefault: 'Token',
         numeric: false,
+        sortable: true,
       },
       {
         id: 'amount',
         name: 'str.amount',
         nameDefault: 'Amount',
         numeric: true,
+        sortable: true,
       },
       {
         id: 'fee',
         name: 'str.fee',
         nameDefault: 'Fee',
         numeric: true,
+        sortable: true,
       },
       {
         id: 'status',
         name: 'str.status',
         nameDefault: 'Status',
         numeric: false,
+        sortable: true,
       },
       {
         id: 'actions',
         name: 'str.actions',
         nameDefault: 'Actions',
         numeric: false,
+        sortable: false,
       },
     ];
 
     return (
       <TableHead>
         <TableRow>
-          {headerCols.map((column) => (
-            <TableCell
-              key={column.id}
-              numeric={column.numeric}
-              sortDirection={orderBy === column.id ? order : false}
-            >
-              <Tooltip
-                title={<FormattedMessage id="str.sort" defaultMessage="Sort" />}
-                enterDelay={Config.intervals.tooltipDelay}
-                placement={column.numeric ? 'bottom-end' : 'bottom-start'}
-              >
-                <TableSortLabel
-                  active={orderBy === column.id}
-                  direction={order}
-                  onClick={this.createSortHandler(column.id)}
-                >
-                  <FormattedMessage id={column.name} default={column.nameDefault} />
-                </TableSortLabel>
-              </Tooltip>
-            </TableCell>
-          ))}
+          {headerCols.map((column) => column.sortable ? this.getSortableCell(column) : this.getNonSortableCell(column))}
         </TableRow>
       </TableHead>
     );
   }
 
-  getTableRows(transactions) {
+  getSortableCell = (column) => {
```

Comment:
Just from experience, this style of splitting the dom into various functions is being moved away from. Instead, putting them in separate components keeps things more concise since dom elements *are* components, not methods. i.e.
```javascript
class EventHistory extends Component {
  ...
  render() {
    return (
      <TableHead>
        <TableRow>
          {headerCols.map((column) => column.sortable ? (
            <SortableTableCell {...column} {...this.state} onClick={this.createSortHandler(column.id)} />
           ) : (
             <NonSortableTableCell {...column} {...this.state} />
           ))}
        </TableRow>
      </TableHead>
    );
  }
}

const SortableTableCell = ({ id, numeric, name, nameDefault, order, orderBy, onClick }) => (
  <TableCell key={id} numeric={numeric} sortDirection={orderBy === id ? order : false}>
    <Tooltip
      title={<FormattedMessage id="str.sort" defaultMessage="Sort" />}
      enterDelay={Config.intervals.tooltipDelay}
      placement={numeric ? 'bottom-end' : 'bottom-start'}
    >
      <TableSortLabel active={orderBy === id} direction={order} onClick={onClick}>
        <FormattedMessage id={name} default={nameDefault} />
      </TableSortLabel>
    </Tooltip>
  </TableCell>
)

const NonSortableTableCell = ({ id, numeric, name, nameDefault, onClick }) => (
  <TableCell key={id} numeric={numeric}>
    <FormattedMessage id={name} default={nameDefault} />
  </TableCell>
)
```

Reply:
i tried doing this but it threw an error with having to define the props for id, numeric, etc..